src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,109 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo },
+      isFound: false
+    };
+  }
+
+  handleDivClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+    for (var file in this.props.svgOptions) {
+      if (this.props.svgOptions[file] === this.props.value) {
+        this.state.isFound = true;
+        this.state.index = file;
+      }
+    }
+    if (!this.state.isFound) {
+      this.props.svgOptions.push(this.props.value);
+    } else {
+      this.props.svgOptions.splice(this.state.index, 1);
+      this.state.isFound = false;
     }
+  };
+
+  toChangePluginObjectValues = arrayWithPluginOptions => {
+    let newSvgoObject = this.props.svgObject.plugins.filter(
+      option => {
+        for (var element of arrayWithPluginOptions) {
+          if (option[element] !== null && option[element] !== undefined) {
+            option[element] = true;
+          }
+          return true;
+        }
+      }
+    );
+
+    return newSvgoObject;
+  };
+
+  changingPluginValuesPerUserOption= () => {
+    this.changePluginObjectValues(this.props.svgOptions);
+  };
 
-	handleDivClick= () => {
-	    this.setState({
-	        isClicked: !this.state.isClicked
-	    });
-	}
-
-	render() {
-	    return (
-	        <div OnClick ={this.handleDivClick}>
-	            <input className='checkbox-input-svg' type='checkbox' checked={this.state.isClicked}
-	            />
-	            <label className='checkbox-label-svg'>
-	                {this.props.option}
-	            </label>
-	        </div>
-	    );
-	}
+  render() {
+    return (
+      <div onClick={this.handleDivClick}>
+        <input type="checkbox" checked={this.state.isClicked} />
+        <label className="checkbox-label-svg">{this.props.option}</label>
+      </div>
+    );
+  }
 }
+
 SvgSetting.propTypes = {
-    option: PropTypes.string,
-    isSelected: PropTypes.bool
+  svgSettingList: PropTypes.object,
+  svgSettingList : PropTypes.object,
```

Comment:
why is the same prop declared twice

Reply:
This is not used anywhere in this file, why is it being extracted from the redux store